protocol/bc/byte32.go

```diff
@@ -0,0 +1,89 @@
+package bc
+
+import (
+	"bytes"
+	"database/sql/driver"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"chain/errors"
+)
+
+type Byte32 [32]byte
```

Comment:
As far as I can tell, this type is mostly only used inside Hash's methods, and it would not be too hard to inline some of these or lightly rearrange things and get rid of this type entirely. How does that sound?